src/angular.headroom.js

```diff
@@ -4,30 +4,30 @@
     return;
   }
   
-  function headroom($document, HeadroomService) {
+  function headroom(HeadroomService) {
     return {
       scope: {
         tolerance: '=',
         offset: '=',
         classes: '=',
         scroller: '@'
       },
-      link: function link($scope, $element) {
+      link: function ($scope, $element) {
         var options = {};
         var opts = HeadroomService.options;
-        if (options.scroller) {
-          options.scroller = $document.querySelector(options.scroller);
-        }
         for (var prop in opts) {
           options[prop] = $scope[prop] || opts[prop];
         }
+        if ($scope.scroller) {
+          options.scroller = document.querySelector($scope.scroller);
+        }
```

Comment:
It needs to be moved. This way we make sure `options.scroller` is a DOM element.Otherwise after the `for` loop `options.scroller` would be equals to `$scope.scroller` which is a string.